witte_jacquelyn_ndvi.py

```diff
@@ -0,0 +1,898 @@
+# 
+# 1. The code to scale if you added new sites or more time periods to the analysis.
+# 2. Someone else to understand your workflow.
+# 3. The LEAST and most efficient (i.e. runs fast, minimize repetition) amount of code that completes the task.
+# 
+# ### HINTS
+# 
+# * Remove values outside of the landsat valid range of values as specified in the metadata, as needed.
+# * Keep any output files SEPARATE FROM input files. Outputs should be created in an outputs directory that is created in the code (if needed) and/or tested for.
+# * Use the functions that we demonstrated during class to make your workflow more efficient.
+# * BONUS - if you  chose - you can export your data as a csv file. You will get bonus points for doing this.
+# 
+# 
+# ## Assignment Requirements
+# 
+# Your submission to the GitHub repository should include:
+# * This Jupyter Notebook file (.ipynb) with:
+#     * The code to create a plot of mean NDVI across a year for  2 NEON Field Sites:
+#         * NDVI on the x axis and formatted dates on the y for both NEON sites on one figure/axis object
+#     * The **data should be cleaned to remove the influence of clouds**. See the [earthdatascience website for an example of what your plot might look like with and without removal of clouds](https://www.earthdatascience.org/courses/earth-analytics-python/create-efficient-data-workflows/).
+# * BONUS: Create one output `.csv` file that has 3 columns - NDVI, Date and Site Name - with values for SJER and HARV.
+# 
+# Your notebook should:
+# * Have *at least* 2 well documented and well named functions with docstrings.
+# * Include a Markdown cell at the top of the notebook that outlines the overall workflow using pseudocode (i.e. plain language, not code)
+# * Include additional Markdown cells throughout the notebook to describe: 
+#     * the data that you used - and where it is from
+#     * how data are being processing
+#     * how the code is optimized to run fast and be more concise
+
+# # Replace this cell with your pseudocode  for this workflow
+# 
+# If you happen to be a diagram person a diagram is ok too
+# 
+# 
+
+# # Pseudocode for this workflow
+- There are three required files to get a single mean NDVI value:
+    1. Landsat tif
+    2. Landsat QC tif
+    3. Landsat shapefile
+
+The basic flow to calculate the mean NDVI for a single date of measurements:
+- Read the image tifs
+- Read the QA tif
+- Read the shapefile
+- Merge the individual tif bands into a single xarray
+- Crop the dataArray using the shapefile
+- Clean the cropped dataArray - remove 0's and high values
+- Mask the dataArray using the QA tif
+- Calculate the NDVI
+- Calculate the mean NDVI
+- Turn it into a dataFrame
+- Plot it
+
+The idea is to create many functions to perform most of the outlined steps above. When looping through all the data directories to calculate a single mean NDVI value and grab it's metadata, the sequence of code will be a series of function calls only.
```

Comment:
@jacquiewitte Psuedo-code is probably in the grey area for PEP8 compliance, but this line may be easier split into separate lines